data-protocols/ids/ids-transform-v1/src/main/java/org/eclipse/dataspaceconnector/ids/transform/type/TransformUtils.java

```diff
@@ -0,0 +1,49 @@
+/*
+ *  Copyright (c) 2022 Fraunhofer Institute for Software and Systems Engineering
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.ids.transform.type;
+
+import org.eclipse.dataspaceconnector.ids.spi.types.IdsId;
+import org.eclipse.dataspaceconnector.ids.spi.types.IdsType;
+import org.eclipse.dataspaceconnector.spi.transformer.TransformerContext;
+
+import java.util.List;
+
+public final class TransformUtils {
+
+    /**
+     * Check if create {@link IdsId} is not null and of the correct type. If not, report problem to context
+     * and ignore (return true) or return false.
+     *
+     * @param context transformer context.
+     * @param idsId value to check.
+     * @param types list of expected types.
+     * @param ignore set to false if missing id should cause malformed object handling
+     * @return valid or not.
+     */
+    public static boolean isInvalidId(TransformerContext context, IdsId idsId, List<IdsType> types, Boolean ignore) {
```

Comment:
This method is pretty hard to interpret, especially the "ignore" field, plus if the idsId is not null the return is always `false`, so this method is not checking the "invalidity", but the "null" state, with the "ignore" variant.
Probably I would inline it, as the `ignore` field is hardcoded and strictly dependent from the caller.
Another option could be to make this returning a `List<String>` that are representing the problems, and then the caller can decide if ignore the fact that the idsId is invalid or not.

Reply:
inligned the code again